server/game/cards/11.6-DitD/SerJasonMallister.js

```diff
@@ -0,0 +1,37 @@
+const DrawCard = require('../../drawcard');
+
+class SerJasonMallister extends DrawCard {
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            condition: () => this.game.getOpponents(this.controller).every(opponent => opponent.shadows.length == 0) && (this.controller.shadows.length == 0),
+            match: this,
+            effect: [
+                ability.effects.addKeyword('Renown'),
+            ]
+        });
+        this.reaction({
+            when: {
+                afterChallenge: event => (
+                    event.challenge.attackingPlayer === this.controller &&
+                    event.challenge.winner === this.controller &&
+                    this.isAttacking()
+                )
+            },
+            target: {
+                activePromptTitle: 'Select a card',
+                cardCondition: card => card.location === 'shadows' && card.controller === this.game.currentChallenge.defendingPlayer &&
+                    this.game.getOpponents(this.controller).every(opponent => opponent.shadows.length !== 0)
```

Comment:
You don't need to check whether all opponents have at least a card in shadows, only whether the defending player has a card in shadows (as done correctly on line 22).